provisioning/baremetal_webhook.go

```diff
@@ -26,17 +29,25 @@ const (
 
 // EnsureBaremetalOperatorWebhook ensures ValidatingWebhook resources are ready to serve.
 func EnsureBaremetalOperatorWebhook(info *ProvisioningInfo) (bool, error) {
-	if !info.BaremetalWebhookEnabled {
-		return false, nil
-	}
-
 	webhookService := newBaremetalOperatorWebhookService(info.Namespace)
 	_, _, err := resourceapply.ApplyService(info.Client.CoreV1(), info.EventRecorder, webhookService)
 	if err != nil {
 		err = errors.Wrap(err, "unable to create validatingwebhook service")
 		return false, err
 	}
 
+	if !info.BaremetalWebhookEnabled {
```

Comment:
We can safely apply webhook service even if BaremetalWebhookEnabled is not enabled(due to service-ca not ready).